superset-frontend/src/SqlLab/components/ResultSet/index.tsx

```diff
@@ -310,12 +320,50 @@ const ResultSet = ({
         schema: query?.schema,
       };
 
-      const canExportData = findPermission(
+      const canExportCsv = findPermission(
         'can_export_csv',
-        'SQLLab',
+        'SqlLab',
+        user?.roles,
+      );
+      const canExportGoogleSheets = findPermission(
+        'can_export_google_sheets',
+        'SqlLab',
         user?.roles,
       );
 
+      // Antd >= 4.24.0 format:
+      const exportMenuItems = [];
+      if (canExportCsv) {
+        exportMenuItems.push({
+          label: t('CSV'),
+          key: 'csv',
+          icon: <Icons.FileOutlined />,
+          onClick: () => {
+            logAction(LOG_ACTIONS_SQLLAB_DOWNLOAD_CSV, {});
+            window.open(getExportCsvUrl(query.id), '_blank')?.focus();
+          },
+        });
+      }
+      if (isFeatureEnabled(FeatureFlag.GoogleSheetsExport) && canExportGoogleSheets) {
+        exportMenuItems.push({
+          label: t('Google Sheets'),
+          key: 'google-sheets',
+          icon: <Icons.GoogleOutlined />,
+          onClick: () =>
+            window.open(getExportGoogleSheetsUrl(query.id), '_blank')?.focus(),
```

Comment:
I hope to add the action logging like the above export csv.

```
```
```suggestion
            logAction(LOG_ACTIONS_SQLLAB_DOWNLOAD_GSHEET, {});
            window.open(getExportGoogleSheetsUrl(query.id), '_blank')?.focus(),
```